### PR TITLE
Fix: Cloud Storage objects are immutable on GCP

### PR DESCRIPTION
**Pull request by maxulysse**
Most likely culprit is the `sort: false` in the `collectFile()`

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test tests/ --verbose --profile +docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by asp8200** [comment]
I guess this should fix [this](https://github.com/nf-core/sarek/issues/1491), right?

Has the proposed solution been tested?

**Review by FriederikeHanssen — APPROVED** [review]
I'll approve, but I think we should get Esha's ok here.

File: subworkflows/local/utils_nfcore_sarek_pipeline/main.nf [hunk 1]
```diff
@@ -280,8 +280,14 @@ def methodsDescriptionText(mqc_methods_yaml) {
...
-    meta["doi_text"] = meta.manifest_map.doi ? "(doi: <a href=\'https://doi.org/${meta.manifest_map.doi}\'>${meta.manifest_map.doi}</a>)" : ""
-    meta["nodoi_text"] = meta.manifest_map.doi ? "": "<li>If available, make sure to update the text to include the Zenodo DOI of version of the pipeline used. </li>"
+    if (meta.manifest_map.doi) {
+        def temp_doi_ref = ""
+        String[] manifest_doi = meta.manifest_map.doi.split(",")
+        for (String doi_ref: manifest_doi) temp_doi_ref += "(doi: <a href=\'https://doi.org/${doi_ref}\'>${doi_ref}</a>), "
+        meta["doi_text"] = temp_doi_ref.substring(0, temp_doi_ref.length()-2)
+    } else {
+        meta["nodoi_text"] = "<li>If available, make sure to update the text to include the Zenodo DOI of version of the pipeline used. </li>"
+    }
```
Comment (maxulysse) [inline subworkflows/local/utils_nfcore_sarek_pipeline/main.nf:290]:
```suggestion
    if (meta.manifest_map.doi) {
        def temp_doi_ref = ""
        String[] manifest_doi = meta.manifest_map.doi.split(",")
        for (String doi_ref: manifest_doi) temp_doi_ref += "(doi: <a href=\'https://doi.org/${doi_ref}\'>${doi_ref}</a>), "
        meta["doi_text"] = temp_doi_ref.substring(0, temp_doi_ref.length()-2)
    } else meta["doi_text"] = ""
    meta["nodoi_text"] = meta.manifest_map.doi ? "" : "<li>If available, make sure to update the text to include the Zenodo DOI of version of the pipeline used. </li>"
```

**Comment by maxulysse** [comment]
cf https://github.com/nf-core/tools/pull/2946 and https://github.com/nf-core/tools/pull/2947

**Comment by maxulysse** [comment]
sorry for fixing two issues in 1 PR, but I encountered the second issue while fixing the first one

**Comment by maxulysse** [comment]
in the MultiQC:
> Data was processed using nf-core/sarek v3.5.0dev (doi: [10.12688/f1000research.16665.2](https://doi.org/10.12688/f1000research.16665.2)), (doi: [10.1093/nargab/lqae031](https://doi.org/10.1093/nargab/lqae031)), (doi: [10.5281/zenodo.3476425](https://doi.org/10.5281/zenodo.3476425)) of the nf-core collection of workflows ([Ewels et al., 2020](https://doi.org/10.1038/s41587-020-0439-x)), utilising reproducible software environments from the Bioconda ([Grüning et al., 2018](https://doi.org/10.1038/s41592-018-0046-7)) and Biocontainers ([da Veiga Leprevost et al., 2017](https://doi.org/10.1093/bioinformatics/btx192)) projects.

instead of:
> Data was processed using nf-core/sarek v3.5.0dev (doi: [10.12688/f1000research.16665.2, 10.1093/nargab/lqae031, 10.5281/zenodo.3476425](https://doi.org/10.12688/f1000research.16665.2, 10.1093/nargab/lqae031, 10.5281/zenodo.3476425)) of the nf-core collection of workflows ([Ewels et al., 2020](https://doi.org/10.1038/s41587-020-0439-x)), utilising reproducible software environments from the Bioconda ([Grüning et al., 2018](https://doi.org/10.1038/s41592-018-0046-7)) and Biocontainers ([da Veiga Leprevost et al., 2017](https://doi.org/10.1093/bioinformatics/btx192)) projects.

On the terminal:
```
If you use nf-core/sarek for your analysis please cite:

* The pipeline
  https://doi.org/10.12688/f1000research.16665.2
  https://doi.org/10.1093/nargab/lqae031
  https://doi.org/10.5281/zenodo.3476425
```

instead of:
```
If you use nf-core/sarek for your analysis please cite:

* The pipeline
  10.12688/f1000research.16665.2, 10.1093/nargab/lqae031, 10.5281/zenodo.3476425
```

**Comment by maxulysse** [comment]
splitting the DOI thingy in another PR as it's becoming to be a bigger part of the PR

**Review by SPPearce — APPROVED** [review]

**Pull request merged**